### ISSUE TITLE
Hope that this will be setup on Colab

### ISSUE THREAD

**Issue by GeorvityLabs** [NVlabs/eg3d#1]
Would be awesome if the Inversion Demo could be set up on colab.
That way we could also try making 3d avatars of ourselfs with just a single input image.

**Comment by natowi** [comment]
Yes, like the one for https://github.com/facebookresearch/pifuhd

**Comment by robbsaber** [comment]
Any update?

**Comment by cantonioupao** [comment]
I think they won't really do this. Please refer to #21

**Comment by osushilover** [comment]
Is there anyone can make this code an colab notebook?

**Comment by neverix** [comment]
I made one, to be released in 3 days

**Comment by GeorvityLabs** [comment]
> I made one, to be released in 3 days

looking forward to it!

**Comment by neverix** [comment]
Beta, let me know if anything breaks! https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ

**Comment by osushilover** [comment]
> ベータ版、何か壊れたら教えてください！https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ

Thank you for making the Colab notebook!
How can I use this notebook to generate novel views images and head around views videos like the one in #18?
I would like to generate them from a single face photo of a particular person.

**Comment by osushilover** [comment]
> Beta, let me know if anything breaks! https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ

Thank you for making the Colab notebook!
How can I use this notebook to generate novel views images and head around views videos like the one in https://github.com/NVlabs/eg3d/issues/18?
I would like to generate them from a single face photo of a particular person.

**Comment by osushilover** [comment]
> Beta, let me know if anything breaks! https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ

Thank you for making the Colab notebook!
How can I use this notebook to generate novel views images and head around views videos like the one in #28 ?
I would like to generate them from a single face photo of a particular person.

**Comment by neverix** [comment]
You should

> > Beta, let me know if anything breaks! https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ
> 
> Thank you for making the Colab notebook! How can I use this notebook to generate novel views images and head around views videos like the one in #28 ? I would like to generate them from a single face photo of a particular person.

You should put that photo into this cell

![image](https://user-images.githubusercontent.com/46641404/180619608-10c4b89b-85f9-4a70-933b-4bb36d6f7271.png)

(just remove the default URL link)

Then tweak focal length and make sure the optimization doesn't break. After that you can export the model and open it in something like Blender. I will probably add normalization and look around views like in #28, thanks for the idea!

**Comment by osushilover** [comment]
> > > Beta, let me know if anything breaks! https://colab.research.google.com/drive/1nu4maHdENDav54UUYk7H0MjC3IBP81yZ
> > 
> > 
> > Thank you for making the Colab notebook! How can I use this notebook to generate novel views images and head around views videos like the one in #28 ? I would like to generate them from a single face photo of a particular person.
> 
> You should put that photo into this cell
> 
> ![image](https://user-images.githubusercontent.com/46641404/180619608-10c4b89b-85f9-4a70-933b-4bb36d6f7271.png)
> 
> (just remove the default URL link)
> 
> Then tweak focal length and make sure the optimization doesn't break. After that you can export the model and open it in something like Blender. I will probably add normalization and look around views like in #28 in the next version, thanks for the idea!

Thank you for your reply.
I see. I'd like to try the around view version.
Once that's done, I want to be first in line to run it!

**Comment by neverix** [comment]
> Thank you for your reply. I see. I'd like to try the around view version. Once that's done, I want to be first in line to run it!

Lookarounds are done #47

**Comment by osushilover** [comment]
> Lookarounds are done #47

I've done, too!
But The generated images do not resemble the person's face, have different faces mixed in, or have a lot of noise.
![exam01](https://user-images.githubusercontent.com/77590366/180692427-921b0b66-d42b-43db-b66e-ee96f1938580.jpg)
https://streamable.com/qw5l4z
And I would like to download "video.mp4" instead of "video.gif" if possible, cause I couldn't have found that mp4.
What should I do?

**Comment by neverix** [comment]
It looks like your optimization broke (look at the upper left corner, it should be similar to upper right). You should tune parameters like `focal_length` and the optimization weights.

You can change the video save path in the cell below from `video.gif` to `video.mp4` but I haven't tested it

**Comment by osushilover** [comment]
That's do a helpful advise!
I'll study Stylegan2 more deeply.

**Comment by luminohope** [comment]
FYI We added additional scripts that can preprocess in-the-wild images compatible with the FFHQ checkpoints.
https://github.com/NVlabs/eg3d/issues/18#issuecomment-1200366872

**Comment by juanjomon** [comment]
> 

Thanks Neverix for your work. It is really appreciated. 
I use your Colab notebook and I feel like the images generated are always frontal view and the colours are usually dull. Why is this? . Also the inversion is not very accurate in terms of likeness. Is this due to the database? , Is there any way to obtain the same results as in the videos of the paper (different positions, full colour) and the same quality in inversion as in Stylegan3?.
It would be great if this could be setup
On the other hand, showing the depth maps is a great feature!.
Thanks again for your work.

**Comment by metekutlu07** [comment]
Thank you very much @neverix for the notebook. Is it possible to use another dataset, for example of flowers ? We are trying to convert images of flowers into basic 3D forms for an artistic project. It seems possible in their explanation but maybe not on the collar notebook ? I am pretty new on the subject. All help is greatly appreciated !

**Comment by jasuriy** [comment]
@neverix hi
were you able to run the code before converting it into colab ?
I was going to train it with my custom dataset so curious about training it?

**Comment by neverix** [comment]
No, I haven't trained my own EG3D model or used the repo outside of Colab.
